x-pack/test/observability_onboarding_api_integration/tests/custom_logs/privileges.spec.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const registry = getService('registry');
+  const observabilityOnboardingApiClient = getService('observabilityOnboardingApiClient');
+
+  async function callApiAsLogMonitoringUser() {
+    return await observabilityOnboardingApiClient.logMonitoringUser({
+      endpoint: 'GET /internal/observability_onboarding/custom_logs/privileges',
+    });
+  }
+
+  async function callApiAsAdminUser() {
+    return await observabilityOnboardingApiClient.adminUser({
+      endpoint: 'GET /internal/observability_onboarding/custom_logs/privileges',
+    });
+  }
+
+  async function callApiAsReadUser() {
+    return await observabilityOnboardingApiClient.readUser({
+      endpoint: 'GET /internal/observability_onboarding/custom_logs/privileges',
+    });
+  }
+
+  async function callApiAsNoAccessUser() {
+    return await observabilityOnboardingApiClient.noAccessUser({
+      endpoint: 'GET /internal/observability_onboarding/custom_logs/privileges',
+    });
+  }
```

Comment:
Is this possible?

```suggestion
  async function callApiAs(user: User) {
    return await observabilityOnboardingApiClient[user]({
      endpoint: 'GET /internal/observability_onboarding/custom_logs/privileges',
    });
  }
```

Reply:
Addressed in [e7b03cd](https://github.com/elastic/kibana/pull/160165/commits/e7b03cd03c779408adeed28275f3241b4c9939ee).